libs/components/src/packing-list/PackingListItem.tsx

```diff
@@ -101,6 +103,49 @@ export const PackingListItem: FunctionComponent<PackingListItemProps> = (props)
   const dispatch = useDispatch()
   const size = useWindowSize()
   const [removing, setRemoving] = useState(false)
+  const [labels, setLabels] = useState<JSX.Element[]>([])
+
+  const labelEntries = Object.entries(props.item.labels || [])
+
+  const handleRemoveLabel = (labelId: string) => {
+    if (!props.item.labels) return
+
+    const tmpLabels: Record<string, any> = { ...props.item.labels }
+    delete tmpLabels[labelId]
+
+    firebase
+      .firestore()
+      .collection('trips')
+      .doc(props.tripId)
+      .collection('packing-list')
+      .doc(props.item.id)
+      .update({
+        labels: {
+          ...tmpLabels,
+        }
+      })
+
+    // toast.success(`${label.text} label added!`)
+  }
+
+  useEffect(() => {
+    setLabels(labelEntries.map(([id, label]) => {
```

Comment:
i would pull out the JSX and have that be in the return down below (where `{labels}` is) and only use the effect to update the labels. Once the labels are updated in state, you can simply map over them the same way you are here, but in the JSX below